terraso_backend/tests/graphql/mutations/test_sites.py

```diff
@@ -394,7 +394,7 @@ def test_site_transfer_unlinked_site_user_viewer_failure(client, user, site, pro
     input_data = {"siteIds": [str(site.id)], "projectId": str(project.id)}
     client.force_login(user)
     payload = graphql_query(SITE_TRANSFER_MUTATION, client=client, input_data=input_data).json()
-    assert "errors" in payload
+    assert match_json("*..badPermissions[*].id", payload) == [str(site.id)]
```

Comment:
The change to the permissions implementation means that this method now has consistent failure modes in all cases (previously it behaved differently based on whether the failed permission change was "add to project" or "transfer site"; now those checks are unified.)